ytsync/sources/youtubeVideo.go

```diff
@@ -103,16 +103,21 @@ func (v YoutubeVideo) download() error {
 		return nil
 	}
 
-	downloader := ytdl.NewYoutube(false)
-	err = downloader.DecodeURL("https://www.youtube.com/watch?v=" + v.id)
+	videoUrl := "https://www.youtube.com/watch?v=" + v.id
+	videoInfo, err := ytdl.GetVideoInfo(videoUrl)
 	if err != nil {
 		return err
 	}
-	err = downloader.StartDownload(videoPath)
+
+	var downloadedFile *os.File
+	downloadedFile, err = os.Create(v.getFilename())
 	if err != nil {
 		return err
 	}
-	return nil
+
+	defer downloadedFile.Close()
+
+	return videoInfo.Download(videoInfo.Formats.Best(ytdl.FormatAudioEncodingKey)[0], downloadedFile)
```

Comment:
you didn't address the issue where `videoInfo.Formats.Best(ytdl.FormatAudioEncodingKey)` may be empty, so it panics when you try to take the first one